File: docker/client/scripts/_ts_lib/database-utils.ts
import postgres from 'postgresjs';

export class DatabaseUtils {
  static async dropTables(sql: ReturnType<typeof postgres>) {
    const result = await sql`
      SELECT tablename FROM pg_tables WHERE schemaname = 'public';
    `;
    for (const row of result) {
      if (row.tablename != 'shared_queries') {
        await sql`DROP TABLE IF EXISTS ${sql(row.tablename)} CASCADE`;
      }
    }
  }

  static async emptyTable(sql: ReturnType<typeof postgres>, tableName: string) {
    await sql`DELETE FROM ${sql(tableName)}`;
  }

  /**
   * Returns a postgres client for accessing the database.
   */
  static openClient() {
    return postgres({
      host: 'pgbouncer-service',
      port: 6432,
      database: 'testdb',
      username: 'user',
      password: 'password',
    });
  }
}

File: bench/src/lib/database_utils.go
package lib

import (
	"context"
	"fmt"

	"github.com/jackc/pgx/v5"
)

func DropTables(conn *pgx.Conn) error {
	rows, err := conn.Query(context.Background(), "SELECT tablename FROM pg_tables WHERE schemaname = 'public'")
	if err != nil {
		return err
	}
	defer rows.Close()

	for rows.Next() {
		var tablename string
		err = rows.Scan(&tablename)
		if err != nil {
			return err
		}

		if tablename != "shared_queries" {
			query := fmt.Sprintf("DROP TABLE IF EXISTS %s CASCADE", tablename)
			_, err = conn.Exec(context.Background(), query)
			if err != nil {
				return err
			}
		}
	}
	return nil
}

func EmptyTable(conn *pgx.Conn, tableName string) error {
	query := fmt.Sprintf("DELETE FROM %s", tableName)
	_, err := conn.Exec(context.Background(), query)
	return err
}
